Sources/Algorithms/Pad.swift

```diff
@@ -0,0 +1,163 @@
+//
+// Copyright (c) 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+extension RangeReplaceableCollection {
+  /// Returns a new collection of the same type whose length is expanded
+  /// to `paddedCount` by repeating the padding `element` at the start.
+  ///
+  /// In case the length of the collection is longer than or equal to the
+  /// `paddedCount`, the original collection is returned.
+  ///
+  /// This example uses `paddingStart(with:toCount:)` to obtain
+  /// `String`s justified to the desired `paddedCount`.
+  ///
+  ///     let numbers = (7...12).map(String.init)
+  ///         .map { $0.paddingStart(with: "0", toCount: 2) }
+  ///     print(numbers) // ["07", "08", "09", "10", "11", "12"]
+  ///
+  /// - Parameters:
+  ///   - element: The element which is repeated to expand the collection.
+  ///   - paddedCount: The length of the padded collection.
+  ///
+  /// - Complexity: O(_m_) when the collection's length is less than the
+  /// `paddedCount`, where _m_ is equal to the `paddedCount`. O(_1_) for
+  /// a `RandomAccessCollection`; otherwise O(_n_) when the collection's length
+  /// is greater than or equal to the `paddedCount`, where _n_ is the length of
+  /// the collection.
+  @inlinable
+  public func paddingStart(
+    with element: Element,
+    toCount paddedCount: Int
+  ) -> Self {
+    let padElementCount = paddedCount - count
+
+    // Early exit if no padding is required.
+    guard padElementCount > 0 else { return self }
+
+    var paddedCollection = Self.init()
+    paddedCollection.reserveCapacity(paddedCount)
+    paddedCollection.append(contentsOf: repeatElement(element, count: padElementCount))
+    paddedCollection.append(contentsOf: self)
+
+    return paddedCollection
+  }
+
+  /// Mutates a `RangeReplaceableCollection` by expanding its length
+  /// to `paddedCount` repeating the padding `element` at the start.
+  ///
+  /// The collection is not mutated in the case where the length of the collection
+  /// is greater than or equal to the `paddedCount`.
+  ///
+  /// This example uses `padStart(with:toCount:)` to get a `String`
+  /// justified to the desired `paddedCount`.
+  ///
+  ///     var myString = "Hello, World!"
+  ///     myString.padStart(with: " ", toCount: 20)
+  ///     myString // "       Hello, World!"
+  ///
+  /// - Parameters:
+  ///   - element: The element which is repeated to expand the `Collection`.
+  ///   - paddedCount: The length of the padded `Collection`.
+  ///
+  /// - Complexity: O(_m_) when the collection's length is less than the
+  /// `paddedCount`, where _m_ is equal to the `paddedCount`. O(_1_) for
+  /// a `RandomAccessCollection`; otherwise O(_n_) when the collection's length
```

Comment:
That wouldn't be `O(m)` given the complexity of `insert(contentsOf:at:)` is `O(m + n)` where n is length of the original collection and m is the length of newElements(which in this case is paddedCount) ?

Reply:
Even if `m` was equal to `n` it would be `O(m + m)` -> `O(2 * m)` -> `O(m)` because constants could be dropped.

Reply:
Ah yeah, you right. It seems like I read it wrong, the statement is correct then =]

Reply:
If we are talking about the case where the `collection.count` >= `paddedCount`, the complexity is O(_n_) because we simply need to compute the `count` and exit early as no padding is required.

Reply:
yeah, the statement is correct is just that I read it wrong the conditions 😅